### PR TITLE
Handle empty or incomplete responses from getos

### LINKED ISSUE #7: Getos returning empty response causes TypeError
Hi there, we've just had a case where a super custom version of Arch Linux caused `linux-distro` to throw a type error: `TypeError: Cannot read property 'dist' of null`.

I can see that there is already a PR for similar case in #6, I'll add one more check there and I hope you guys will merge it.

### PR DESCRIPTION
**Pull request by jakubzitny**
I've picked up #6 by @paulcbetts, dropped one commit and added one more check.

Closes #7.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -19,11 +19,14 @@ module.exports = () => {
 		};
 	}).catch(() => {
 		return pify(getos).then(res => {
+			if (!res) {
+				res = { };
```
Comment (kevva) [inline index.js]: I rather throw an error.

Comment (jakubzitny) [inline index.js]: @kevva It still can be a "valid" response from `getos`, can't it? They don't support everything. I am trying to get rid of having to wrap my calls to `linux+distro` try-catch. But I guess I should do it anyway, right?

Comment (kevva) [inline index.js]: If `res` is `undefined` we should throw an error with the message `Your distro couldn't be determined`, or something along those lines. And yes, if you don't want your application to crash you should do:

```
linuxDistro()
  .then(data => console.log(data))
  .catch(() => {
    // ignore error
  });
```

**Comment by jakubzitny** [comment]
@kevva updated.

File: index.js (hunk 1 above)
Comment (jakubzitny) [inline index.js]: @kevva Wanna merge this, or do you want other things fixuped?

**Comment by vojtatranta** [comment]
@kevva could you please merge it?

**Comment by kevva** [comment]
Fixed in https://github.com/kevva/linux-distro/commit/a5b07cf043c2750e58ec538f2b7a7f927993ab7a.

**Pull request closed without merging**